packages/esm-patient-chart-app/src/visit/visit-form/visit-form.test.tsx

```diff
@@ -197,24 +199,59 @@ describe('Visit form', () => {
     expect(screen.getByText(/Inpatient Ward/i)).toBeInTheDocument();
   });
 
-  it('renders an error message when a visit type has not been selected', async () => {
+  it('renders a validation error when required fields are not filled', async () => {
     const user = userEvent.setup();
 
     renderVisitForm();
 
     const saveButton = screen.getByRole('button', { name: /start visit/i });
-    const locationPicker = screen.getByRole('combobox', { name: /Select a location/i });
+    const locationPicker = screen.getByRole('combobox', { name: /select a location/i });
     await user.click(locationPicker);
     await user.click(screen.getByText('Inpatient Ward'));
-
     await user.click(saveButton);
 
-    expect(screen.getByText(/Missing visit type/i)).toBeInTheDocument();
-    expect(screen.getByText(/Please select a visit type/i)).toBeInTheDocument();
+    expect(screen.getByText(/missing visit type/i)).toBeInTheDocument();
+    expect(screen.getByText(/please select a visit type/i)).toBeInTheDocument();
 
     await user.click(screen.getByLabelText(/Outpatient visit/i));
   });
 
+  it('displays an error message when the visit start date is in the future', async () => {
+    const user = userEvent.setup();
+
+    renderVisitForm();
+
+    const dateInput = screen.getByRole('textbox', { name: /date/i });
+    const futureDate = dayjs().add(1, 'month').format('DD/MM/YYYY');
+
+    await user.clear(dateInput);
+    await user.type(dateInput, futureDate);
+    await user.tab();
+
+    expect(screen.getByText(/start date needs to be on or before/i)).toBeInTheDocument();
+  });
+
+  // TODO: Figure out why this test is failing
+  xit('displays an error message when the visit start time is in the future', async () => {
```

Comment:
I don't know why this test is failing. It mimics the user interaction of filling in the field with a value that ought to trigger the validation (similar to the test just above).